dialogue-serde/src/test/java/com/palantir/conjure/java/dialogue/serde/EndpointErrorsConjureBodySerDeTest.java

```diff
@@ -0,0 +1,185 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.dialogue.serde;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.common.collect.ImmutableList;
+import com.palantir.conjure.java.api.errors.CheckedServiceException;
+import com.palantir.conjure.java.api.errors.ErrorType;
+import com.palantir.conjure.java.dialogue.serde.EndpointErrorTestUtils.ConjureError;
+import com.palantir.conjure.java.dialogue.serde.EndpointErrorTestUtils.TypeReturningStubEncoding;
+import com.palantir.conjure.java.serialization.ObjectMappers;
+import com.palantir.dialogue.BodySerDe;
+import com.palantir.dialogue.DeserializerArgs;
+import com.palantir.dialogue.TestResponse;
+import com.palantir.dialogue.TypeMarker;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.Safe;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.Unsafe;
+import com.palantir.logsafe.UnsafeArg;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import javax.annotation.processing.Generated;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class EndpointErrorsConjureBodySerDeTest {
+    private static final ObjectMapper MAPPER = ObjectMappers.newServerObjectMapper();
+    private ErrorDecoder errorDecoder = ErrorDecoder.INSTANCE;
+
+    @Generated("by conjure-java")
+    private sealed interface EndpointReturnBaseType permits StringReturn, ErrorForEndpoint {}
+
+    @Generated("by conjure-java")
+    record StringReturn(String value) implements EndpointReturnBaseType {
+        @JsonCreator
+        public static StringReturn create(String value) {
+            return new StringReturn(Preconditions.checkArgumentNotNull(value, "value cannot be null"));
+        }
+    }
+
+    @Generated("by conjure-java")
+    @JsonDeserialize(using = ErrorForEndpoint.Deserializer.class)
+    record ErrorForEndpoint(
+            @Safe String errorCode,
+            @Safe String errorName,
+            @Safe String errorInstanceId,
+            @Safe String arg,
```

Comment:
- can put the args into a separate record which includes matches the `parameters` map for deserialization so we don't need a custom deserializer.